test/gtest/common/sort.cpp

```diff
@@ -0,0 +1,75 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "xhp-tests.hpp"
+
+using T = int;
+using DV = xhp::distributed_vector<T>;
+using LV = std::vector<T>;
+
+void test_sort(LV &v, auto func) {
+  auto size = v.size();
+  DV d_v(size);
+
+  for (std::size_t idx = 0; idx < size; idx++) {
+    d_v[idx] = v[idx];
+  }
+  barrier();
+
+  std::sort(v.begin(), v.end(), func);
+  xhp::sort(d_v, func);
+
+  barrier();
+  EXPECT_TRUE(equal(v, d_v));
+}
+
+void test_sort2s(LV &v) {
+  test_sort(v, std::less<T>());
+  test_sort(v, std::greater<T>());
+}
+
+void test_sort_randomvec(std::size_t size, std::size_t bound = 100) {
+  LV l_v = generate_random<T>(size, bound);
+  test_sort2s(l_v);
+}
+
+TEST(Sort, Random) {
+  test_sort_randomvec(1);
+  test_sort_randomvec(comm_size - 1);
+  test_sort_randomvec((comm_size - 1) * (comm_size - 1));
+  test_sort_randomvec(17);
+  test_sort_randomvec(123);
+}
+
+TEST(Sort, BigRandom) { test_sort_randomvec(1234567, 65535); }
+
+TEST(Sort, NonRandom) {
+  LV v;
+  v = {1, 1, 1, 1, 1, 1, 1, 1, 1, 1};
+  test_sort2s(v);
```

Comment:
maybe simply `test_sort2s({1, 1, 1, 1, 1, 1, 1, 1, 1, 1})`?

Reply:
test_sort2s takes arg by reference, it would be too many copies if I changed this, too

Reply:
It will not. It's a just a unit test. I guess nobody will even be able to measure time difference if you will take things by value in all places.

In unit tests simplicity is so important (reader doesn't need to think who modifies object, test is not polluted by `const &`, `&`, `&&` and similar things), that taking by value should be default way of writing functions, unless one _observes_ that taking by ref really speeds up the test which is too long. Then we should change by-valu to by-ref.

Reply:
Removing big_random test you suggested then. It took too long time anyway, and it was not clear what it tested, comparing to other tests.